libs/api/database/src/lib/content/stores/tags.store.ts

```diff
@@ -22,6 +22,25 @@ export class TagsStore {
         return this.tags.find({ kind: kind }).sort({ name: 1 });
     }
 
+    /**
+     * Returns all tags of a specified kind, sorted by parent (none first) then alphabetically.
+     * @param kind
+     */
+    async fetchTagsSortedByParent(kind: TagKind): Promise<TagsDocument[]> {
+        return this.tags.find({ kind: kind }).sort({ parent: 1, name: 1 });
+    }
+
+    /**
+     * Returns all tags of a specified kind without parents, sorted alphabetically.
+     * @param kind
+     */
+    async fetchParentTags(kind: TagKind): Promise<TagsDocument[]> {
+        return this.tags.find({
+            kind: kind,
+            parent: null,
```

Comment:
Warning, this might be a bit of a sneaky performance hole, because `parent` isn't indexed. This query will force Mongo to scan every tag in the collection.

...and in fact, now that I'm looking at this, `find({kind:kind})` might be too, for the same reason--neither of these fields have indexes.

...we should really do a bit of profiling here.